Experiments_toolkit.py

```diff
@@ -0,0 +1,190 @@
+            percent_train -- double ([0-1] Percentage of data to go to training set)
+            test -- dict (Testing dictionary )
+            percent_test -- double (0-1] Percentage of data to go to testing set)
+            validation -- dict (Validation dictionary )
+            percent_validation -- double ([0-1] Percentage of data to go to validation set)
+
+        Return:
+            train -- dict (Dictionary with percent_train of list)
+            test -- dict (Dictionary with percent_test of list)
+            validation -- dict (Dictionary with percent_validation of list)
+        """
+    length = len(data)
+    train_index = int(length * percent_train)
+    test_index = int(length * percent_test)
+    train[key] = data[0:train_index]
+    test[key] = data[train_index:train_index + test_index]
+    validation[key] = data[train_index + test_index:]
+    return train, test, validation
+
+
+def preprocess_data(data, rows, filtering_params, window_function, window_size, offset):
+    """
+       Description: Process the samples stored in data by windowing the traces and saving
+                                these windows (numpy arrays) into a dictionary.
+
+       Args:
+            data -- list (List of numpy arrays where the last row is time steps, 
+            			  [(sensor_dims + 1), num_samples])
+            rows -- int (Number of rows in the data)
+            filtering_params -- All_Filtering_Params (Dataclass found in windowing.py that 
+                                stores important parameters for the 2 steps involved with 
+                                finding the precise start and end time of a activity in the data)
+            window_size -- int (Window size of the trace to be saved into the
+            				    numpy array windows that will be returned)
+            offset -- int (Number to add to the predicted start time of the 
+            			   activity from where the window will start at)
+
+       Return:
+           windows-- int[] (Array of the windows cropped by the window_function)
+           """
+    windows = []
+    for sample in data:
+        activity_windows = wl.initial_find_activity(
+            sample, rows, filtering_params.initial_filtering.window_size,
+            filtering_params.initial_filtering.stride,
+            filtering_params.initial_filtering.divisor)
+        if len(activity_windows) > 0:
+            start_time, end_time = wl.get_start_time(
+                activity_windows, rows, filtering_params.precise_filtering.window_size,
+                filtering_params.precise_filtering.stride,
+                filtering_params.precise_filtering.threshold)
+            windowed_trace = window_function(
+                sample, start_time + offset, window_size)
+            if len(windowed_trace[0][0]) != window_size:
+                continue
+            windows.extend(windowed_trace)
+    np_windows = np.array(windows)
+    return np_windows
+
+
+def get_np_X_Y(x, y, length):
```

Comment:
Could you make the function title more descriptive? get_np_X_Y reads like you're extracting two generic numpy arrays. Instead you're creating an array and labels. Maybe combine_data_using_labels? Or something similar?